modules/core/pom.xml

```diff
@@ -73,6 +73,10 @@
       <groupId>org.apache.hadoop</groupId>
       <artifactId>hadoop-client</artifactId>
     </dependency>
+    <dependency>
+      <groupId>org.apache.thrift</groupId>
+      <artifactId>libthrift</artifactId>
```

Comment:
I think this still needs to be `<scope>provided</scope>`, or perhaps `<optional>true</optional>` in order to prevent transitive dependency resolution by consumers of the fluo-core jar. That is, unless the shade plugin also modifies the POM to exclude libthrift... and I'm not sure it does.

Reply:
I found that `<scope>provided</scope>` would exclude it from the shaded jar so I went with `<optional>true</optional>`

Also the shaded plugin doesn't modify the original POM

Reply:
This works, and it probably avoids us having to do anything with the dependency reduced POM... although I would do a comparison with the dependency reduced version manually... just to make sure nothing strange was overlooked.

Reply:
I didn't see thrift in the reduced POM and it changed the checkstyle plugin config

Reply:
It changed a few things... interesting things... but not important things. The reduced POM looks like our POM, so that's good. I'll check the jar contents manually just to do a quick sanity check, but so far, everything looks the way I'd expect.